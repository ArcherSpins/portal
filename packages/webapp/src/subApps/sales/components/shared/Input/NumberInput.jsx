// @flow
import React, { Component } from 'react';
import {
  Container,
  NumberInputWrap,
} from './styled';

type Props = {
  value?: any,
  onChange: (SyntheticInputEvent<HTMLInputElement>) => void,
  inputStyle?: Object,
  style?: Object
}

class NumberInput extends Component<Props> {
  static defaultProps = {
    value: '',
    inputStyle: {},
    style: {},
  }

  onInputChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
    const { onChange } = this.props;
    onChange(e);
  }

  render() {
    const {
      inputStyle,
      style,
    } = this.props;
    return (
      <Container style={style}>
        <NumberInputWrap
          type="number"
          {...this.props}
          style={inputStyle}
          onChange={this.onInputChange}
        />
      </Container>
    );
  }
}

export default NumberInput;
