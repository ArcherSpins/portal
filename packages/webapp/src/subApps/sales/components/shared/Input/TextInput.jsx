// @flow
import React, { Component } from 'react';
import {
  Container,
  InputWrap,
  Label,
  Error,
} from './styled';

type Props = {
  value?: string,
  placeholder: string,
  error?: string | null,
  styling?: 'borderless' | 'default',
  onChange: (SyntheticInputEvent<HTMLInputElement>) => any,
  small?: boolean,
  label?: string,
  style?: Object,
  inputStyle?: Object,
  sideLabel?: string,
}

class TextInput extends Component<Props> {
  static defaultProps = {
    value: '',
    error: null,
    styling: 'default',
    small: false,
    label: '',
    style: {},
    sideLabel: '',
    inputStyle: {},
  }

  onInputChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
    const { onChange } = this.props;
    onChange(e);
  }

  render() {
    const {
      label, error, style, inputStyle,
    } = this.props;
    return (
      <div style={style}>
        <Container>
          {label && <Label>{label}</Label>}
          <InputWrap
            {...this.props}
            style={inputStyle}
            onChange={this.onInputChange}
          />
          <Error>{error}</Error>
        </Container>
      </div>
    );
  }
}

export default TextInput;
