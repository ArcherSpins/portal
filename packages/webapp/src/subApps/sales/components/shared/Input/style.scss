input::-webkit-input-placeholder { color: #AAAAAA; }
input::-moz-placeholder { color: #AAAAAA; }