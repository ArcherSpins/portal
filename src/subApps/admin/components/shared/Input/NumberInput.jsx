// @flow
import React, { Component } from 'react';
import {
  Container,
  NumberInputWrap,
} from './styled';

type Props = {
  value?: any,
  onChange: (SyntheticEvent<HTMLInputElement>) => void,
  inputStyle?: {
    [string]: string | number
  },
  style?: {
    [string]: string | number
  }
}

class NumberInput extends Component<Props> {
  static defaultProps = {
    value: '',
    inputStyle: {},
    style: {},
  }

  onInputChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
    const { onChange } = this.props;
    onChange(e);
  }

  render() {
    const {
      inputStyle,
      style,
    } = this.props;
    return (
      <Container style={style}>
        <NumberInputWrap
          type="number"
          {...this.props}
          style={inputStyle}
          onChange={this.onInputChange}
        />
      </Container>
    );
  }
}

export default NumberInput;
