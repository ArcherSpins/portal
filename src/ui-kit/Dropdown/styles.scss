
.drwn__wrap {
  .drwn__control {
    min-height: 2rem !important;
  }
}
