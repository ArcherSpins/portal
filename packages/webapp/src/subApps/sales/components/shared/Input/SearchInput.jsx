// @flow
import React, { Component } from 'react';
import {
  SearchWrap,
  SearchInput,
  SearchIcon,
} from './styled';
// $FlowFixMe
import img from './search.svg';
// import xmark from './xmark.svg'; Reset, ResetIcon,
import './style.scss';


type Props = {
  placeholder: string,
  error?: boolean,
  styling?: 'borderless' | 'default',
  value?: string,
  onChange: (string, string) => void,
  small?: boolean
}

type State = {
  focused: boolean
}
class Input extends Component<Props, State> {
  static defaultProps = {
    error: false,
    styling: 'default',
    value: '',
    small: false,
  }

  constructor(props: Props) {
    super(props);
    this.state = {
      focused: false,
    };
  }


  onFocus = () => {
    this.setState({
      focused: true,
    });
  }

  onBlur = () => {
    this.setState({
      focused: false,
    });
  }

  reset = () => {
    const { onChange } = this.props;
    onChange('', '');
  }

  onInputChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
    const { onChange } = this.props;
    onChange(e.target.name, e.target.value);
  }

  render() {
    const { focused } = this.state;
    const {
      small, value, onChange, placeholder,
    } = this.props;
    return (
      <div>
        <SearchWrap focused={focused} small={small}>
          <SearchIcon src={img} alt="search" />
          <SearchInput
            {...this.props}
            value={value}
            onFocus={this.onFocus}
            onBlur={this.onBlur}
            onChange={onChange}
            placeholder={placeholder || 'Search..'}
          />
          {/* <Reset onClick={this.reset}>
            <ResetIcon src={xmark} alt="reset" />
          </Reset> */}
        </SearchWrap>
      </div>
    );
  }
}

export default Input;
