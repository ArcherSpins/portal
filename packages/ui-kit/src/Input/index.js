// @flow
import React, { type Node, useState } from 'react';
import MaskedInput from 'react-text-mask';
import noop from 'lodash.noop';
import classNames from 'classnames';
import xmark from './xmark.svg';
import styles from './Input.module.scss';

type ValueType = 'password' | 'text';

type InputType = 'default' | 'borderless';

type InputSize = 'sm' | 'md';

export type InputProps = {
  /** Тип текстового значения инпута. @example: input[type="text"] */
  type?: ValueType,
  placeholder?: string,
  error?: boolean,
  /** Тип инпута */
  use?: InputType,
  disabled?: boolean,
  className?: string,
  value?: string,
  label?: string,
  name: string,
  icon?: Node,
  clearable?: boolean,
  prefix?: string,
  onClearClick?: () => void,
  /** onChange */
  onChange?: (e: SyntheticInputEvent<HTMLInputElement>) => void,
  onFocus?: (e: SyntheticInputEvent<HTMLInputElement>) => void,
  /** Срабатывает при потере фокуса */
  onBlur?: (e: SyntheticInputEvent<HTMLInputElement>) => void,
  mask?: Array<mixed>,
  size?: InputSize
}

const Input = ({
  type,
  placeholder,
  onChange,
  error,
  disabled,
  value,
  className = '',
  name,
  use = 'default',
  label,
  onBlur,
  clearable,
  prefix,
  onClearClick,
  icon,
  mask,
  size,
  onFocus,
  ...restProps
}: InputProps) => {
  const [focused, toggleFocused] = useState(false);

  const handleBlur = (event: SyntheticInputEvent<HTMLInputElement>) => {
    toggleFocused(false);

    if (onBlur) {
      onBlur(event);
    }
  };

  const handleFocus = (event: SyntheticInputEvent<HTMLInputElement>) => {
    toggleFocused(true);

    if (onFocus) {
      onFocus(event);
    }
  };

  if (icon && prefix) throw new Error('icon and prefix props cannot be used at the same time');

  return (
    <div
      className={
        classNames(
          styles[use],
          styles[size],
          {
            [styles.error]: error,
            [styles.paddingLeft]: icon,
            [styles.focus]: focused,
            [styles.disabled]: disabled,
          },
          className,
        )
      }
    >
      <label htmlFor={name}>
        {label}
      </label>
      <div className={classNames(styles.wrap)}>
        {icon && <span className={styles.icon}>{icon}</span>}
        {prefix && <span className={styles.prefix}>{prefix}</span>}
        {
          mask ? (
            <MaskedInput
              mask={mask}
              placeholder={placeholder}
              onChange={onChange}
              onFocus={handleFocus}
              onBlur={handleBlur}
              id={name}
              {...restProps}
              render={(ref, props) => (
                <input
                  ref={ref}
                  {...restProps}
                  name={name}
                  type={type}
                  placeholder={placeholder}
                  disabled={disabled}
                  value={value}
                  {...props}
                />
              )}
            />
          ) : (
            <input
              {...restProps}
              name={name}
              type={type}
              placeholder={placeholder}
              onChange={onChange}
              onFocus={handleFocus}
              onBlur={handleBlur}
              disabled={disabled}
              value={value}
            />
          )
        }
        {/* TODO: REPLACE SVG TO <i /> */}
        {clearable && (value && value.length) && <button onClick={onClearClick} className={styles.clear} type="button"><img src={xmark} alt="clear" /></button>}
      </div>
    </div>
  );
};

Input.defaultProps = {
  type: 'text',
  placeholder: '',
  onChange: noop,
  onBlur: noop,
  onFocus: noop,
  error: false,
  disabled: false,
  use: 'default',
  value: '',
  className: '',
  label: '',
  clearable: false,
  onClearClick: noop,
  icon: null,
  prefix: '',
  mask: null,
  size: 'md',
};

export default Input;
