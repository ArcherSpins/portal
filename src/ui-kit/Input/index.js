// @flow
import React, { type Node } from 'react';
import noop from 'lodash.noop';
import classNames from 'classnames';
import styles from './Input.module.scss';

type ValueType = 'password' | 'text';

type InputType = 'default' | 'borderless';

type Props = {
  /** Тип текстового значения инпута. @example: input[type="text"] */
  type?: ValueType,
  placeholder?: string,
  error?: boolean,
  /** Тип инпута */
  use?: InputType,
  disabled?: boolean,
  className?: string,
  value?: string,
  /** onChange */
  onChange?: (e: SyntheticEvent<HTMLInputElement>) => void,
}

const Input = ({
  type,
  placeholder,
  onChange,
  error,
  disabled,
  value,
  className = '',
  use = 'default',
  ...restProps
}: Props): Node => (
  <div
    {...restProps}
    className={classNames(styles[use], { [styles[error]]: error }, className)}
  >
    <input
      type={type}
      placeholder={placeholder}
      onChange={onChange}
      disabled={disabled}
      value={value}
    />
  </div>
);

Input.defaultProps = {
  type: 'text',
  placeholder: '',
  onChange: noop,
  error: false,
  disabled: false,
  use: 'default',
  value: '',
  className: '',
};

export default Input;
