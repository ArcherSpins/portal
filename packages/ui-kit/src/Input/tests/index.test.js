// TODO: TDD
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import React from 'react';

import styles from '../Input.module.scss';
import Input from '..';


describe('The Input component', () => {
  const onChange = jest.fn();
  it('default input renders correctly', () => {
    const component = <Input />;
    const wrapper = shallow(component);
    expect(wrapper.hasClass(styles.default)).toBeTruthy();
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('borderless input renders correctly', () => {
    const component = <Input use="borderless" />;
    const wrapper = shallow(component);
    expect(wrapper.hasClass(styles.borderless)).toBeTruthy();
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should place value in input', () => {
    const component = <Input use="borderless" value="test" />;
    const wrapper = shallow(component);
    expect(wrapper.find('input').props().value).toEqual('test');
  });

  it('should call onChange', () => {
    const component = <Input onChange={onChange} />;
    const wrapper = shallow(component);
    const event = {
      target: {
        value: 'test',
      },
    };
    wrapper.find('input').simulate('change', event);
    expect(onChange).toHaveBeenCalledWith(event);
  });
});
