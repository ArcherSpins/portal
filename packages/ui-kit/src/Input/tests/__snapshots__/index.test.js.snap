// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Input component borderless input renders correctly 1`] = `
<div
  className="borderless md"
>
  <label>
    
  </label>
  <div
    className="wrap"
  >
    
    <input
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder=""
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`The Input component default input renders correctly 1`] = `
<div
  className="default md"
>
  <label>
    
  </label>
  <div
    className="wrap"
  >
    
    <input
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder=""
      type="text"
      value=""
    />
  </div>
</div>
`;
